src/DataStructures/Tensor/Expressions/AddSubtract.hpp

```diff
@@ -269,8 +205,9 @@ struct AddSub<T1, T2, ArgsList1<Args1...>, ArgsList2<Args2...>, Sign>
                     std::is_same<T2, NumberAsExpression>::value,
                 "Cannot add or subtract Tensors holding different data types.");
   static_assert(
-      detail::AddSubIndexCheck<typename T1::index_list, typename T2::index_list,
-                               ArgsList1<Args1...>, ArgsList2<Args2...>>::value,
+      detail::IndexPropertyCheck<typename T1::index_list,
```

Comment:
Maybe add some `#ifdef SPECTRE_DEBUG` around these sort of checks to reduce compile times in release build? Not for this PR, just something to consider moving forward. (I don't know if these checks have a significant impact on compile times)